src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```
@@ -0,0 +1,1388 @@
+                Message.get("AlkylStructureFragmenter.fragmentSaturationSetting.displayName"));
+        this.keepNonFragmentableMoleculesSetting = new SimpleBooleanProperty(this, "Keep non-fragmentable molecules in Pipeline",
+                AlkylStructureFragmenter.KEEP_NON_FRAGMENTABLE_MOLECULES_SETTING_DEFAULT);
+        this.settingNameTooltipTextMap.put(this.keepNonFragmentableMoleculesSetting.getName(),
+                Message.get("AlkylStructureFragmenter.keepNonFragmentableMoleculesSetting.tooltip"));
+        this.settingNameDisplayNameMap.put(this.keepNonFragmentableMoleculesSetting.getName(),
+                Message.get("AlkylStructureFragmenter.keepNonFragmentableMoleculesSetting.displayName"));
+        this.fragmentSideChainsSetting = new SimpleBooleanProperty(this, "Fragment linear chains",
+                AlkylStructureFragmenter.FRAGMENT_SIDE_CHAINS_SETTING_DEFAULT);
+        this.settingNameTooltipTextMap.put(this.fragmentSideChainsSetting.getName(),
+                Message.get("AlkylStructureFragmenter.fragmentSideChainsSetting.tooltip"));
+        this.settingNameDisplayNameMap.put(this.fragmentSideChainsSetting.getName(),
+                Message.get("AlkylStructureFragmenter.fragmentSideChainsSetting.displayName"));
+        this.maxChainLengthSetting = new SimpleIntegerProperty(this, "Limit length of returned chains",
+                AlkylStructureFragmenter.MAX_CHAIN_LENGTH_SETTING_DEFAULT) {
+            @Override
+            public void set(int newValue) throws NullPointerException, IllegalArgumentException {
+                try {
+                    //call to super.set() for parameter checks
+                    //only allows values greater than 0 to be set
+                    if (newValue > 0) {
+                        super.set(newValue);
+                    }
+                } catch (NullPointerException | IllegalArgumentException anException) {
```

It should throw an exception if the value is below 0.